python/TestHarness/schedulers/QueueManager.py

```diff
@@ -54,17 +54,64 @@ def handleQueueLaunch(self, tester, output):
     def handleQueueStatus(self, tester, output):
         return
 
-    # Return a list of tuples of test files to run
-    # This needs to return an absolute path to a directory containing 'tests' file
-    # along with the 'test' file:
-    # [(dirpath, tests), (dirpath, tests)]
+    # return Prereqs for a given test that may exist in queue_data
+    # returns None type if value does not exist
+    def getPrereqs(self, test_name):
+        return self.getData(self.getJobName(test_name), prereq_tests=True)['prereq_tests']
+
+    # Update dependency information in the queue file
+    def updateDependencies(self, tester):
+        return
+
+    # Get specified data from queue_data
+    def getData(self, job, **kwargs):
+        tmp_dict = {}
+        if job in self.queue_data.keys():
+            for kkey in kwargs.keys():
+                try:
+                    tmp_dict[kkey] = self.queue_data[job][kkey]
+                except KeyError: # Requested key not available
+                    tmp_dict[kkey] = None
+        return tmp_dict
+
+    # Set queue_data (this overwrites attributes)
+    def putData(self, job, **kwargs):
+        for key, value in kwargs.iteritems():
+            if job in self.queue_data.keys():
+                self.queue_data[job][key] = value
+            else:
+                self.queue_data[job] = {key : value}
+
+    # Update queue_data (appends specified values)
+    def updateData(self, job, **kwargs):
+        data = self.getData(job, **kwargs)
+        data_copy = data.copy()
+        for key, value in kwargs.iteritems():
+            tmp_dict = { key : value }
+
+            # Nothing to update, treat as new data (putData)
+            if data[key] != None:
+
+                # There is current data we need to append to
+                if type(data[key]) == type([]) and type(value) == type([]):
+                    data_copy[key].extend(value)
+                    self.putData(job, **data_copy)
```

Comment:
as we continue to modify data_copy, we can probably do putData all in one go after the for loop.